torch/onnx/symbolic_helper.py

```diff
@@ -1598,11 +1580,14 @@ def quantize_helper(
         )
 
     assert scale is not None
-    if scale.type().scalarType() != "Float":
+    if _type_utils.JitScalarType.from_value(scale) != _type_utils.JitScalarType.FLOAT:
```

Comment:
Behavior change. SymbolicValueError or ValueError will be raised when `.scalarType()` returns None.

Reply:
`scale` comes from torcj JIT core, so it is a value `torch._C.Value` - there is a `beartype` + type annotation to add to this assumption, plus an `assert scale is not None` statement. Given that, `from_value` guarantees a valid `JitScalarType` will be returned. Do you agree?

Reply:
Yea I agree `scale` is a good `torch._C.Value`. But a good `torch._C.Value` can have `type` that doesn't have `ScalarType` no? Maybe not for this case, since `scale` is likely constant, but other similar lines where previously a direct comparison between `.scalarType()` and expected type.

Reply:
discussed offline. refactoring symbolics to make sure intended type fallback is done

although we didn't catch an "empty" `type().scalarType()`, this is an easy fix and we may not have aught due to missing unit tests for it